### PR TITLE
feat: Complaint registration and SLA tracking (#231)

### LINKED ISSUE #231: Implement: Klachtenregistratie (Complaint Registration) — Delta Spec
## OpenSpec Change: klachtenregistratie

Spec path: `openspec/changes/klachtenregistratie`

Generated from Specter (3 features).

### PR DESCRIPTION
**Pull request by rubenvdlinde**
Closes #231

## Summary
Implements the klachtenregistratie feature for Pipelinq, providing complaint registration, categorization, and SLA-based deadline tracking. Complaints are linked to contacts and organizations with full audit trail of status changes and a background job for SLA monitoring.

## Spec Reference
- Issue: #231
- Spec: `openspec/changes/klachtenregistratie/design.md`

## Changes
- `lib/Service/ComplaintSlaService.php` — SLA deadline calculation and monitoring service
- `lib/BackgroundJob/ComplaintSlaJob.php` — Timed background job for SLA deadline checks
- `tests/Unit/Service/ComplaintSlaServiceTest.php` — Unit tests for SLA service (7 test methods)
- `tests/Unit/BackgroundJob/ComplaintSlaJobTest.php` — Unit tests for background job (3 test methods)
- `src/views/complaints/` — Frontend Vue components for complaint management
- `src/views/clients/ClientDetail.vue` — Integrated complaints section on client detail view

## Test Coverage
- `tests/Unit/Service/ComplaintSlaServiceTest.php` — SLA calculation, deadline checks, edge cases
- `tests/Unit/BackgroundJob/ComplaintSlaJobTest.php` — Job execution and configuration validation
- Frontend component integration tested via complaint list, detail, and client detail views

### CODE REVIEW

File: lib/Controller/HealthController.php [hunk 1]
```diff
@@ -109,7 +109,7 @@ private function checkDatabase(): string
             return 'ok';
         } catch (\Exception $e) {
             $this->logger->error('[HealthController] Database check failed', ['error' => $e->getMessage()]);
-            return 'failed: '.$e->getMessage();
+            return 'failed';
```
Comment (rubenvdlinde) [inline lib/Controller/HealthController.php:112]: [fixed: exception detail stripped from HTTP response] Rule: CWE-209 / OWASP A05:2021 — checkDatabase() returned 'failed: ' + $e->getMessage() in the health-check JSON, exposing internal DB error strings (e.g. SQLSTATE codes, host names) to all authenticated users. Fixed: return 'failed'; error detail remains in the server log via $this->logger->error().

File: lib/Controller/HealthController.php [hunk 2]
```diff
@@ -131,7 +131,7 @@ private function checkFilesystem(): string
 
             return 'ok';
         } catch (\Exception $e) {
-            return 'failed: '.$e->getMessage();
+            return 'failed';
```
Comment (rubenvdlinde) [inline lib/Controller/HealthController.php:134]: [fixed: exception detail stripped from HTTP response] Rule: CWE-209 / OWASP A05:2021 — checkFilesystem() returned 'failed: ' + $e->getMessage() in the health-check JSON, potentially exposing temp-directory paths or OS error strings to all authenticated users. Fixed: return 'failed'; error detail remains in server logs.

File: lib/Controller/ProspectController.php [hunk 3]
```diff
@@ -106,7 +106,7 @@ public function createLead(): JSONResponse
             return new JSONResponse(data: $result, statusCode: 201);
         } catch (\Exception $e) {
             return new JSONResponse(
-                data: ['error' => $e->getMessage()],
+                data: ['error' => 'An internal error occurred'],
```
Comment (rubenvdlinde) [inline lib/Controller/ProspectController.php:109]: [fixed: exception detail stripped from HTTP response] Rule: CWE-209 / OWASP A05:2021 — createLead() returned $e->getMessage() in its 500 response, accessible to all authenticated users (@NoAdminRequired). Internal exception details (service names, stack traces) should not reach clients. Fixed: generic 'An internal error occurred'; details remain in server logs.

File: lib/Controller/ContactSyncController.php [hunk 4]
```diff
@@ -68,7 +68,7 @@ public function search(): JSONResponse
         } catch (\Exception $e) {
             return new JSONResponse(
                     [
-                        'error' => $e->getMessage(),
+                        'error' => 'An internal error occurred',
```
Comment (rubenvdlinde) [inline lib/Controller/ContactSyncController.php:71]: [fixed: exception detail stripped from HTTP response] Rule: CWE-209 / OWASP A05:2021 — this ContactSyncController method returned $e->getMessage() in its 500 JSON response to @NoAdminRequired users. Service names, internal identifiers, or stack-trace fragments could be exposed. Fixed: generic 'An internal error occurred'; details remain in server logs.

File: lib/Controller/ContactSyncController.php [hunk 5]
```diff
@@ -109,7 +109,7 @@ public function import(): JSONResponse
...
```
Comment (rubenvdlinde) [inline lib/Controller/ContactSyncController.php:112]: [fixed: exception detail stripped from HTTP response] Rule: CWE-209 / OWASP A05:2021 — this ContactSyncController method returned $e->getMessage() in its 500 JSON response to @NoAdminRequired users. Service names, internal identifiers, or stack-trace fragments could be exposed. Fixed: generic 'An internal error occurred'; details remain in server logs.

File: lib/Controller/ContactSyncController.php [hunk 6]
```diff
@@ -150,7 +150,7 @@ public function writeBack(): JSONResponse
...
```
Comment (rubenvdlinde) [inline lib/Controller/ContactSyncController.php:153]: [fixed: exception detail stripped from HTTP response] Rule: CWE-209 / OWASP A05:2021 — this ContactSyncController method returned $e->getMessage() in its 500 JSON response to @NoAdminRequired users. Service names, internal identifiers, or stack-trace fragments could be exposed. Fixed: generic 'An internal error occurred'; details remain in server logs.

**Comment by rubenvdlinde** [comment]
## Security Review — Clyde Barcode

**Result: PASS** (6 fixed, 1 unfixed WARNING, 4 unfixed SUGGESTION)

---

### Scan summary

| Check | Result |
|---|---|
| composer audit | ✅ No packages to audit |
| npm audit --production | ⚠️ Drift signal (see below) |
| gitleaks | ⚠️ Skipped (orchestrator skip) |
| semgrep p/security-audit + p/owasp-top-ten + p/secrets | ✅ 0 findings on all changed PHP files |
| OWASP manual diff review | ✅ Completed |

---

### Fixed findings (6)

**[WARNING] CWE-209 — Exception message disclosure in HealthController** `lib/Controller/HealthController.php:112,134`  
`checkDatabase()` and `checkFilesystem()` returned `'failed: '.$e->getMessage()` in the health-check JSON response, exposing internal DB error strings (SQLSTATE codes, host names) and OS error details to all authenticated users. **Fixed:** return `'failed'`; details remain in server logs.

**[WARNING] CWE-209 — Exception message disclosure in ProspectController** `lib/Controller/ProspectController.php:109`  
`createLead()` returned `$e->getMessage()` in its 500 response, accessible to all `@NoAdminRequired` users. **Fixed:** generic `'An internal error occurred'`.

**[WARNING] CWE-209 — Exception message disclosure in ContactSyncController** `lib/Controller/ContactSyncController.php:71,112,153`  
`search()`, `import()`, `writeBack()` all returned `$e->getMessage()` in 500 JSON to `@NoAdminRequired` users. **Fixed:** generic `'An internal error occurred'` in all three.

---

### Unfixed findings

**[unfixed: requires API integration testing] [WARNING] CWE-598 — KVK API key passed as URL query parameter** `lib/Service/KvkApiClient.php` (~line 119)  
`fetchResults()` builds the KVK API URL with `'apikey' => $apiKey` inside `http_build_query()`. The key appears verbatim in all server-side access logs and proxy logs. Recommended fix: move the key to an HTTP request header (`'apikey' => $apiKey` in the Guzzle `headers` option) — the KVK Handelsregister API supports header-based authentication. Escalated; requires end-to-end testing against the live endpoint before applying.

**[unfixed: admin-only endpoints, SUGGESTION] CWE-209 — Exception message disclosure in admin controllers**  
The following admin-only methods expose `$e->getMessage()` in error responses (impact is limited to admins who typically have log access):
- `lib/Controller/SettingsController.php:177` — `reimport()` returns `$e->getMessage()` on 500
- `lib/Controller/RequestChannelController.php:83,106,128` — `create()`, `update()`, `destroy()`
- `lib/Controller/LeadSourceController.php:83,106,128` — same pattern
- `lib/Controller/ProspectSettingsController.php:131` — `update()` returns `$e->getMessage()` on 500  
These lines are in unchanged hunks of changed files; not fixed in this pass. Recommend addressing in a follow-up.

---

### npm audit drift signal

Pre-review quality gate reported `npm-audit` as **PASSED**, but the current scan finds **16 vulnerabilities (2 HIGH, 5 moderate, 9 low)** — notably:
- `fast-xml-parser` — stack overflow + entity expansion bypass (HIGH)
- `minimatch` — ReDoS via multiple vulnerability chains (HIGH)
- `dompurify` — multiple XSS variants (moderate)

`package.json` / `package-lock.json` are NOT in this PR's changed files, so these are pre-existing. The drift between the quality gate's pass and the current scan may indicate the quality gate uses `--audit-level=high` or filters differently. Recommend triaging these separately.

---

See inline comments on the diff for per-fixed-finding detail.

**Comment by rubenvdlinde** [comment]
Closed as part of the 2026-05-10 hydra pipeline backlog drain (second pass). The underlying spec lives at openspec/changes/<spec-name>/ on `development` and has drifted from this PR's content since it was opened. If still wanted, the next hydra pipeline run will regenerate fresh from the current spec.
